From 45c0216857eeaac386d17dc62e7686ca1deee266 Mon Sep 17 00:00:00 2001
From: David Hanina <dhanina@redhat.com>
Date: Tue, 18 Feb 2025 09:42:43 +0100
Subject: [PATCH 13/16] Warn missing fips file

Also renamed  test_fips_no_fips_enabled to test_fips_no_fips_available
as this name is more fitting, meaning the kernel is missing fips.

Signed-off-by: David Hanina <dhanina@redhat.com>
---
 src/ipahealthcheck/meta/core.py | 5 +++--
 tests/test_meta.py              | 6 +++---
 2 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/src/ipahealthcheck/meta/core.py b/src/ipahealthcheck/meta/core.py
index 6743ab1..d9e4c49 100644
--- a/src/ipahealthcheck/meta/core.py
+++ b/src/ipahealthcheck/meta/core.py
@@ -28,8 +28,9 @@ class MetaCheck(Plugin):
         rval = constants.SUCCESS
         if not os.path.exists(paths.PROC_FIPS_ENABLED):
             fips = "missing {}".format(paths.PROC_FIPS_ENABLED)
-            logger.debug("Can't find %s, skipping" %
-                         paths.PROC_FIPS_ENABLED)
+            logger.warning("Can't find %s, skipping" %
+                           paths.PROC_FIPS_ENABLED)
+            rval = constants.WARNING
         else:
             try:
                 proc_fips_enable_path = Path(paths.PROC_FIPS_ENABLED)
diff --git a/tests/test_meta.py b/tests/test_meta.py
index 7ad0843..9a86373 100644
--- a/tests/test_meta.py
+++ b/tests/test_meta.py
@@ -35,7 +35,7 @@ def gen_result(returncode, output='', error=''):
 
 class TestMetaFIPS(BaseTest):
     @patch('os.path.exists')
-    def test_fips_no_fips_enabled(self, mock_exists):
+    def test_fips_no_fips_available(self, mock_exists):
         mock_exists.return_value = False
 
         framework = object()
@@ -47,7 +47,7 @@ class TestMetaFIPS(BaseTest):
         assert len(self.results) == 1
 
         result = self.results.results[0]
-        assert result.result == constants.SUCCESS
+        assert result.result == constants.WARNING
         assert result.source == 'ipahealthcheck.meta.core'
         assert result.check == 'MetaCheck'
         assert result.kw.get('fips') == 'missing %s' % paths.PROC_FIPS_ENABLED
@@ -195,7 +195,7 @@ class TestMetaACME(BaseTest):
         assert len(self.results) == 1
 
         result = self.results.results[0]
-        assert result.result == constants.SUCCESS
+        assert result.result == constants.WARNING
         assert result.source == 'ipahealthcheck.meta.core'
         assert result.check == 'MetaCheck'
         assert result.kw.get('acme') == \
-- 
2.48.1

